### ISSUE TITLE
vim: if the focus is on the multibuffer search input, Ctrl+o opens the file dialog instead of going back

### ISSUE THREAD

**Issue by thorbenk** [zed-industries/zed#16017]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

In vim mode, I'm typing `g/` to open the project-wide search and type some query. Sometimes, I need to go back to my code to figure out exactly what I should search for :sweat_smile: , so I type `Ctrl-O` while the focus is still in the search box.
This will pop up the "Open file dialog".

If I instead hit enter after formulating my search query, `Ctrl-O` takes me back to my previous file as expected.

### Environment

Linux with vim mode enabled.

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by thorbenk** [comment]
Still an issue.

**Comment by dinocosta** [comment]
@thorbenk Are you still able to reproduce this? Just tried and I've managed to get back to the editor using `ctrl-o`, while being focused in the project search's query input (`g/` in vim mode) 🙂

**Comment by thorbenk** [comment]
I tried with 4dcfe0cff9 and I _can_ still reproduce this. 'g/', 'Ctrl+O', then system file open dialog opens.

**Comment by dinocosta** [comment]
> I tried with [4dcfe0c](https://github.com/zed-industries/zed/commit/4dcfe0cff90a5f5eb1f24190add29a52b0c19a76) and I _can_ still reproduce this. 'g/', 'Ctrl+O', then system file open dialog opens.

Ah, my bad! Silly me forgot to double-check which OS you were using, this doesn't happen on MacOS because the shortcut to open the file dialog is `cmd-o`, so it doesn't conflict with `ctrl-o`.

This simply seems to be a consequence of how the [default Linux keymap](https://github.com/zed-industries/zed/blob/main/assets/keymaps/default-linux.json) is set, where `ctrl-o` is the default binding for `workspace::Open`, regardless of context. I guess the only current way to work around it would be to update your [user keymap](https://zed.dev/docs/key-bindings#user-keymaps) with something like below:

```json
[
  {
    "bindings": {
      "ctrl-o": "pane::GoBack"
    }
  },
  {
    "context": "ProjectPanel",
    "bindings": {
      "cmd-o": "workspace::Open"
    }
  }
]
```

This would make it so that pressing `ctrl-o` would always behave like Vim's `ctrl-o`, regardless of context, but when focused on the Project Panel (`ctrl-shift-e` on Linux) pressing `ctrl-o` would open the file dialog instead.

Hope that helps a little bit!

**Comment by 5brian** [comment]
Technically ctrl-o and ctrl-i shouldn't work on macos either, since it doesn't match the context     `"context": "VimControl && !menu",`

Something about menu bar keybinds triggering in different contexts

![Image](https://github.com/user-attachments/assets/17cd7980-80d0-4bfe-a7f5-6d6a4bf2a6d6)

**Comment by dinocosta** [comment]
> Technically ctrl-o and ctrl-i shouldn't work on macos either, since it doesn't match the context `"context": "VimControl && !menu",`
> 
> Something about menu bar keybinds triggering in different contexts
> 
> ![Image](https://github.com/user-attachments/assets/17cd7980-80d0-4bfe-a7f5-6d6a4bf2a6d6)

Yep, I think you're right, just noticed that when I do use `ctrl-o` while focused on the project's search bar the "Go" menu in the status bar flashes and I believe it's because it's just activating the `Go > Back` action, as you shared, well spotted!

**Comment by thorbenk** [comment]
> Ah, my bad! Silly me forgot to double-check which OS you were using, this doesn't happen on MacOS because the shortcut to open the file dialog is `cmd-o`, so it doesn't conflict with `ctrl-o`.
> 

Thank you, I just tried your bindings and it works 😃

**Comment by ConradIrwin** [comment]
Having played with this a bit, I think the current behavior is probably right. And given that it's easy to change if you want with bindings, let's not change this.

If we did want ctrl-o to go back from project search, then we'd want ctrl-i to refocus the search bar, which it currently doesn't do (unless you have no search). It also is consistent that vim shortcuts only work in vim-enabled editors right now.
